### PR TITLE
adsp: power: make needlessly global assembly functions static

### PR DESCRIPTION
**Pull request by lyakh**
C functions called from assembly aren't recognised by the compiler as being called, so if they're `static` the compiler complains. Make them `used` explicitly

### CODE REVIEW

**Comment by lyakh** [comment]
I just realised, that this PR *effectively* reverts https://github.com/lyakh/zephyr/commit/6a0b1da158a4f8bc5f2ca5058637dce26a38660e but not completely, so if this is confirmed as the right approach, a better fix would indeed be just reverting that commit

**Comment by marc-hb** [comment]
> I just realised, that this PR effectively reverts https://github.com/lyakh/zephyr/commit/6a0b1da158a4f8bc5f2ca5058637dce26a38660e but not completely

This was @RanderWang's commit. Tagging him in the comments as he cannot be added as a reviewer.

**Comment by andyross** [comment]
Just a drive-by comment: this kind of mismatch and thinko will continue to happen as long as intel_adsp insists on having its own synchronous suspend and context save API.  If you just suspended out of the idle thread like the rest of the universe does, this kind of bug couldn't happen.  I understand that there's a need for host involvement, but that can be as simple as a bitmask next to FW_STATUS indicating "This Core Has Reached Idle"[1], and another somewhere for "The Firmware Wants This Core Alive".  The kernel ISR then checks those two bit sets at each interrupt and starts/stops cores as directed.

Basically: this platform is carrying a ton of needless complexity to solve problems the OS already solves for it.

[1] Which is racy, of course, and will require a short timeout on the kernel side to be sure the core is indeed idle.  But that race exists anyway if you try to restart a core from the host kernel immediately after shutting it down.

**Comment by marc-hb** [comment]
SOF test results unfortunately don't look good:
https://github.com/thesofproject/sof/pull/8823#issuecomment-1920119666

Mostly fail on the NOCODEC test configurations, others PASS.

File: soc/xtensa/intel_adsp/ace/power.c [hunk 1]
```diff
@@ -179,7 +179,7 @@ void power_gate_entry(uint32_t core_id)
 	z_xt_ints_off(0xffffffff);
 }
 
-void power_gate_exit(void)
+static void __attribute__((used)) power_gate_exit(void)
```
Comment (nashif) [inline soc/xtensa/intel_adsp/ace/power.c:182]: we have __used for `__attribute__((__used__))`

Comment (lyakh) [reply soc/xtensa/intel_adsp/ace/power.c]: @nashif thanks, fixed

**Review by dcpleung — DISMISSED** [review]

**Review by peter-mitsis — DISMISSED** [review]

**Comment by lyakh** [comment]
no code change, a rebase to adapt to the new tree layout

**Review by dcpleung — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]

**Review by nashif — APPROVED** [review]
